cli/transport.go

```diff
@@ -38,7 +38,9 @@ func (m minCachedTransport) RoundTrip(req *http.Request) (*http.Response, error)
 		return nil, err
 	}
 
-	if resp.Header.Get("expires") == "" && !strings.Contains(resp.Header.Get("cache-control"), "max-age") {
+	// Automatically cache for the minimum time if the request is successful and
+	// the response doesn't already have cache headers.
+	if resp.StatusCode < 400 && resp.Header.Get("expires") == "" && !strings.Contains(resp.Header.Get("cache-control"), "max-age") {
```

Comment:
The only thing I'd say about this line is that it feels long for an if-clause; I might recommend a small helper function that takes the header and status code and returns the appropriate boolean.